ssl/ssl_cert.c

```diff
@@ -664,7 +665,8 @@ STACK_OF(X509_NAME) *SSL_load_client_CA_file_ex(const char *file,
     ret = NULL;
  done:
     /* restore the old libctx */
-    OSSL_LIB_CTX_set0_default(prev_libctx);
+    if (prev_libctx != NULL)
```

Comment:
Shouldn't this be `if (libctx != NULL)` here?

Reply:
No. `prev_libctx == NULL && libctx != NULL` can happen only if OSSL_LIB_CTX_set0_default() fails which means the prev_libctx is not meaningful anyway. And there are code paths which can get here where the OSSL_LIB_CTX_set0_default() was not called at all.

Reply:
Logically I find this strange. In my mind the condition for checking whether to set a new default lib ctx should be the same for deciding whether to reset it on exit. It is odd to use different conditions IMO.

Either way, should we be testing for prev_libctx being NULL after the first `OSSL_LIB_CTX_set0_default()` and erroring out in that case?

Reply:
> Either way, should we be testing for prev_libctx being NULL after the first `OSSL_LIB_CTX_set0_default()` and erroring out in that case?

Yes, that would make sense. But this condition should be still kept.